include/librealsense2/dds/dds-device-broadcaster.cpp

```diff
@@ -223,35 +218,37 @@ bool dds_device_broadcaster::create_device_writer( const std::string &device_key
     // data_sharing for this topic, which we did here.
     // (See https://github.com/eProsima/Fast-DDS/issues/2641)
 
-    //wqos.data_sharing().automatic();
+    // wqos.data_sharing().automatic();
     wqos.data_sharing().off();
     //---------------------------------------------------------------------------------------
 
-    wqos.ownership().kind = EXCLUSIVE_OWNERSHIP_QOS;
-    std::shared_ptr< dds_client_listener > writer_listener
-        = std::make_shared< dds_client_listener >( this );
+    std::shared_ptr< dds_client_listener > writer_listener = std::make_shared< dds_client_listener >( this );
 
-    _device_handle_by_sn[device_key]
-        = { rs2_device,
-            _publisher->create_datawriter( _topic, wqos, writer_listener.get() ),
-            writer_listener };
+    _device_handle_by_sn[device_key] = { rs2_device,
+                                         _publisher->create_datawriter( _topic, wqos, writer_listener.get() ),
+                                         writer_listener };
 
     return _device_handle_by_sn[device_key].data_writer != nullptr;
 }
 
 void dds_device_broadcaster::create_broadcast_topic()
 {
-    // Registering the topic type enables topic instance creation by factory
+    // Topic constructor creates TypeObject that will be sent as part of the discovery phase
+    librealsense::dds::topics::raw::device_info raw_msg;
+
     eprosima::fastdds::dds::TypeSupport topic_type( new librealsense::dds::topics::device_info::type );
+    // Auto fill type object but not type information so sniffer can dynamically match a reader for this topic
```

Comment:
I didn't understand from the comment what is the difference between 
`auto_fill_type_information` and `auto_fill_type_object` and whay one needs to be on and one off,
Can you elaborate do if we get here later in the development we will understand the logic?

Reply:
I will elaborate a bit in the comment

`TypeObject` And `TypeInformation` are terms defined in the [DDS-XTypes](https://www.omg.org/spec/DDS-XTypes/1.3/PDF) standard.
`TypeObject` contains the full information about the type, needed to dynamically build it (at the remote endpoint, e.g the sniffer). `TypeInformation` is an approach to reduce discovery traffic, sending lighter identifiers, and only interested `DomainParticipants` will request the full `TypeObject` (using additionally built domain entities).

Since we **are** interested in the `TypeObject`, using the `TypeInformation` is redundant and wasteful.
In current Fast-DDS implementation if both `TypeInformation` and `TypeObject` are sent the subscriber to the topic will not be matched (probably need to use additional services like `TypeLookupService` to match)

Reply:
Thanks for the info, I would have updated the comment with some point from this explanation (Not all..)
Something like : 
We turn on `TypeObject` because ... 
We turn off `TypeInformation ` because ...

Reply:
Done